### PR TITLE
Add GitHub Action to automate labeling on issues and pull requests

### LINKED ISSUE #3026: [FR] Introduce a bot that automatically applies labels on issues and pull requests
Thank you for submitting a feature request. **Before proceeding, please review MLflow's [Issue Policy for feature requests](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md#feature-requests) and the [MLflow Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst)**.

**Please fill in this feature request template to ensure a timely and thorough response.**

## Willingness to contribute
The MLflow Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature (either as an MLflow Plugin or an enhancement to the MLflow code base)?

- [x] Yes. I can contribute this feature independently.
- [ ] Yes. I would be willing to contribute this feature with guidance from the MLflow community.
- [ ] No. I cannot contribute this feature at this time.

## Proposal Summary

I'm developing a bot using GitHub Actions that automatically applies labels on issues and pull requests by detecting task lists in the description.

https://github.com/harupy/auto-labeling

## Motivation
- What is the use case for this feature?
- Why is this use case valuable to support for MLflow users in general?
- Why is this use case valuable to support for your project(s) or organization?
- Why is it currently difficult to achieve this use case? (please be as specific as possible about why related MLflow features and components are insufficient)

### What component(s), interfaces, languages, and integrations does this feature affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interfaces
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Languages 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

## Details

(Use this section to include any additional information about the feature. If you have a proposal for how to implement this feature, please include it here. For implementation guidelines, please refer to the [Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#contribution-guidelines).)

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Add GitHub Action to automates labeling on issues and pull requests (resolves #3026).

## How is this patch tested?

Tested on [my repo](https://github.com/harupy/auto-labeling).

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/auto-labeling.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+on:
+  schedule:
+    # Run this action every 10 minutes
+    - cron: "*/10 * * * *"
```
Comment (harupy) [inline .github/workflows/auto-labeling.yml:4]: To apply labels on an issue or a pull request, a write permmision is required, but the GitHub token created for a pull request coming from a forked repo doesn't have a write permission. Instead, we can use the `schdule` event and trigger this action on a certain interval.

File: .github/workflows/auto-labeling.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+jobs:
+  labeling:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: harupy/auto-labeling@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          label-pattern: '- \[(.*?)\] ?`(.+?)`'
```
Comment (harupy) [inline .github/workflows/auto-labeling.yml:14]: What this action does is:

1. Fetch issues and pull requests (created within a month past)
2. For each issue or pull request, extract labels from the description and add (or remove) them.

Comment (smurching) [reply .github/workflows/auto-labeling.yml:16]: @harupy awesome! Do you have an idea of how many Github REST API requests the action makes? Is it proportional to the number of issues & PRs created within the past month? Would just be good to sanity-check that we won't run into the [REST API limits on Github actions](https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits):

> You can execute up to 1000 API requests in an hour across all actions within a repository. If exceeded, additional API calls will fail, which might cause jobs to fail. This limit also applies to self-hosted runners.

Comment (smurching) [reply .github/workflows/auto-labeling.yml:16]: If it seems like we'll hit the threshold, maybe we can run the action less frequently, e.g. every hour?

**Review by smurching — COMMENTED** [review]
One comment/question, otherwise LGTM!

File: .github/workflows/auto-labeling.yml (hunk 2 above)
Comment (harupy) [reply .github/workflows/auto-labeling.yml:16]: It is proportional to the number of issues & PRs & labels that need to be updated. Given the threshold, every-10-minutes is too frequent. I'll fix it to every-hour as you suggested.

File: README.rst [hunk 3]
```diff
@@ -16,14 +16,17 @@ currently run ML code (e.g. in notebooks, standalone applications or the cloud).
...
 
 .. |docs| image:: https://img.shields.io/badge/docs-latest-success.svg
     :target: https://mlflow.org/docs/latest/index.html
     :alt: Latest Docs
 .. |travis| image:: https://img.shields.io/travis/mlflow/mlflow.svg
     :target: https://travis-ci.org/mlflow/mlflow
     :alt: Build Status
+.. |labeling| image:: https://github.com/mlflow/mlflow/workflows/Labeling/badge.svg
+    :target: https://github.com/mlflow/mlflow/actions?query=workflow%3ALabeling
+    :alt: Labeling Action Status
```
Comment (harupy) [inline README.rst:29]: Added a status badge for this action to make it easier to notice issues.

**Review by smurching — APPROVED** [review]
LGTM

**Pull request merged**